pxtcompiler/emitter/service.ts

```diff
@@ -1254,11 +1256,18 @@ namespace ts.pxtc.service {
                     if (si.snippetAddsDefinitions
                         || (isPython && !si.pySnippet)
                         || (!isPython && !si.snippet)) {
-                        let { snippet, addsDefinitions } = getSnippet(lastApiInfo.apis, takenNames, v.runtime, si, n, isPython);
-                        if (isPython)
+                        const snippetNode = getSnippet(lastApiInfo.apis, takenNames, v.runtime, si, n, isPython);
+                        const snippet = snippetStringify(snippetNode)
+                        const snippetWithMarkers = snippetStringify(snippetNode, true)
+                        const addsDefinitions = snippetAddsDefinitions(snippetNode)
+                        if (isPython) {
                             si.pySnippet = snippet
-                        else
+                            si.snippetWithMarkers = snippetWithMarkers
+                        }
+                        else {
                             si.snippet = snippet
+                            si.snippetWithMarkers = snippetWithMarkers
+                        }
```

Comment:
```suggestion
                        }
                        else {
                            si.snippet = snippet
                        }
                        si.snippetWithMarkers = snippetWithMarkers
```

maybe pull out ` si.snippetWithMarkers = snippetWithMarkers` to be outside of the conditional as it's the same in both cases?

Reply:
Oops it's supposed to be different in each case, thanks!